packages/grid/x-data-grid/src/models/events/gridEventLookup.ts

```diff
@@ -271,6 +271,15 @@ export interface GridControlledStateEventLookup {
   columnVisibilityModelChange: { params: GridColumnVisibilityModel };
 }
 
+export interface GridControlledStateReasonLookup {
+  filter:
+    | 'addFilterItem' // TODO v6: Merge with upsertFilterItem
+    | 'upsertFilterItem'
```

Comment:
The "Add filter" button is a bit hacky. It doesn't use `apiRef.current.upsertFilterItem`, but instead directly updates the model. I think we should improve this by using the API method only.

Reply:
Can't we use `apiRef.current.upsertFilterItem` there instead of `apiRef.current.setFilterModel` and not wait for v6?

Reply:
I tried, but then it broke too many tests and a gave up. When the "Add filter" button is clicked, it first adds the new item to an internal state, then only when the value is changed it updates the model. Maybe we need a `upsertFilterItems` receiving more than one item. I won't merge this PR until we agree about the reasons.

Reply:
I think first filter is added to the local state when filter panel opens. Then clicking "Add filter" just adds one more filter, and then both filters are added to the filter state.
We can either make `upsertFilterItems` receive an array of items, or call it multiple times (and then `onFilterModelChange` will be called multiple times as well)